### PR TITLE
Add min-juju-version support to charm-proof. Fixes #141.

### LINKED ISSUE #141: charm proof needs min-juju-version support
Please add min-juju-version support to charm proof in preparation for the feature landing.  This will allow us to add the metadata to development versions of charms and pass proof.  TIA

### PR DESCRIPTION
**Pull request by tvansteenburgh**
Fixes #141

### CODE REVIEW

File: charmtools/charms.py [hunk 1]
```diff
@@ -500,6 +502,30 @@ def schema_type(self, **kw):
...
+
+    :param charm: dict of charm metadata parsed from metadata.yaml
+    :param linter: :class:`CharmLinter` object to which info/warning/error
+        messages will be written
+
+    """
+    if 'min-juju-version' not in charm:
+        return
+
+    pattern = r'^(\d{1,9})\.(\d{1,9})(\.|-(\w+))(\d{1,9})(\.\d{1,9})?$'
```
Comment (marcoceppi) [inline charmtools/charms.py]: couldn't we just do
`pattern = r'^(\d{2,9})\.(\d{1,9})(\.|-(\w+))(\d{1,9})(\.\d{1,9})?$'`

Comment (marcoceppi) [inline charmtools/charms.py]: nevermind, better error handling with this.

Comment (tvansteenburgh) [inline charmtools/charms.py]: Well we could do `\d[2-9]{1,9}`, but my point with the original version was to give them a more specific error msg if the format was okay, but the version wasn't high enough. If you think that's not important I'll change it.

**Comment by marcoceppi** [comment]
LGTM :boat:

File: charmtools/charms.py (hunk 1 above)
Comment (tvansteenburgh) [inline charmtools/charms.py]: Heh, ninja'd.

**Pull request merged**